File: Task310/blog/src/main/java/com/blog/controller/TagController.java
package com.blog.controller;

import com.blog.dto.TagRequestTo;
import com.blog.dto.TagResponseTo;
import com.blog.service.TagService;
import jakarta.validation.Valid;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping("/api/v1.0/tags")
public class TagController {

    private final TagService tagService;

    public TagController(TagService tagService) {
        this.tagService = tagService;
    }

    @GetMapping
    public ResponseEntity<List<TagResponseTo>> getAllTags() {
        List<TagResponseTo> tags = tagService.findAll();
        return ResponseEntity.ok(tags);
    }

    @GetMapping("/{id}")
    public ResponseEntity<TagResponseTo> getTagById(@PathVariable Long id) {
        TagResponseTo tag = tagService.findById(id);
        return ResponseEntity.ok(tag);
    }

    @PostMapping
    public ResponseEntity<TagResponseTo> createTag(@Valid @RequestBody TagRequestTo request) {
        TagResponseTo createdTag = tagService.create(request);
        return ResponseEntity.status(HttpStatus.CREATED).body(createdTag);
    }

    @PutMapping("/{id}")
    public ResponseEntity<TagResponseTo> updateTag(@PathVariable Long id, @Valid @RequestBody TagRequestTo request) {
        TagResponseTo updatedTag = tagService.update(id, request);
        return ResponseEntity.ok(updatedTag);
    }

    @DeleteMapping("/{id}")
    public ResponseEntity<Void> deleteTag(@PathVariable Long id) {
        tagService.deleteById(id);
        return ResponseEntity.noContent().build();
    }
}

File: 351002/Melnikov/publisher/controllers/tag_controller.py
from fastapi import APIRouter, Response, status

from ..dto.tag_dto import TagRequestTo, TagResponseTo
from ..services.tag_service import TagService

router = APIRouter(prefix="/api/v1.0/tags", tags=["tag"])
service = TagService()


@router.post("", response_model=TagResponseTo, status_code=status.HTTP_201_CREATED)
def create_tag(dto: TagRequestTo):
    return service.create(dto)


@router.get("", response_model=list[TagResponseTo], status_code=status.HTTP_200_OK)
def get_tags():
    return service.get_all()


@router.get("/{tag_id}", response_model=TagResponseTo, status_code=status.HTTP_200_OK)
def get_tag(tag_id: int):
    return service.get_by_id(tag_id)


@router.put("/{tag_id}", response_model=TagResponseTo, status_code=status.HTTP_200_OK)
def update_tag(tag_id: int, dto: TagRequestTo):
    return service.update(tag_id, dto)


@router.delete("/{tag_id}", status_code=status.HTTP_204_NO_CONTENT)
def delete_tag(tag_id: int):
    service.delete(tag_id)
    return Response(status_code=status.HTTP_204_NO_CONTENT)